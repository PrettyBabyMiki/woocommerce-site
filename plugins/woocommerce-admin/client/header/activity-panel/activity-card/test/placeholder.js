/**
 * External dependencies
 */
import { shallow } from 'enzyme';

/**
 * Internal dependencies
 */
import { ActivityCardPlaceholder } from '../';

describe( 'ActivityCardPlaceholder', () => {
	test( 'should render a default placeholder', () => {
		const card = shallow( <ActivityCardPlaceholder /> );
		expect( card ).toMatchSnapshot();
	} );

	test( 'should render a card placeholder with subtitle placeholder', () => {
		const card = shallow( <ActivityCardPlaceholder hasSubtitle /> );
		expect( card ).toMatchSnapshot();
	} );

	test( 'should render a card placeholder with date placeholder', () => {
		const card = shallow( <ActivityCardPlaceholder hasDate /> );
		expect( card ).toMatchSnapshot();
	} );

	test( 'should render a card placeholder with action placeholder', () => {
		const card = shallow( <ActivityCardPlaceholder hasAction /> );
		expect( card ).toMatchSnapshot();
	} );

	test( 'should render a card placeholder with all optional placeholder', () => {
		const card = shallow(
			<ActivityCardPlaceholder hasAction hasDate hasSubtitle />
		);
		expect( card ).toMatchSnapshot();
	} );

	test( 'should render a card placeholder with multiple lines of content', () => {
		const card = shallow( <ActivityCardPlaceholder lines={ 3 } /> );
		expect( card ).toMatchSnapshot();
	} );

	test( 'should render a card placeholder with no content', () => {
		const card = shallow( <ActivityCardPlaceholder lines={ 0 } /> );
		expect( card ).toMatchSnapshot();
	} );
} );
