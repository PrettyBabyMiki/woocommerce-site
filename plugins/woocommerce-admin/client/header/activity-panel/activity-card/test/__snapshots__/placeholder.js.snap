// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActivityCardPlaceholder should render a card placeholder with action placeholder 1`] = `
<div
  aria-hidden={true}
  className="woocommerce-activity-card is-loading"
>
  <span
    className="woocommerce-activity-card__icon"
  >
    <span
      className="is-placeholder"
    />
  </span>
  <div
    className="woocommerce-activity-card__header"
  >
    <div
      className="woocommerce-activity-card__title is-placeholder"
    />
  </div>
  <div
    className="woocommerce-activity-card__body"
  >
    <span
      className="is-placeholder"
      key="0"
    />
  </div>
  <div
    className="woocommerce-activity-card__actions"
  >
    <span
      className="is-placeholder"
    />
  </div>
</div>
`;

exports[`ActivityCardPlaceholder should render a card placeholder with all optional placeholder 1`] = `
<div
  aria-hidden={true}
  className="woocommerce-activity-card is-loading"
>
  <span
    className="woocommerce-activity-card__icon"
  >
    <span
      className="is-placeholder"
    />
  </span>
  <div
    className="woocommerce-activity-card__header"
  >
    <div
      className="woocommerce-activity-card__title is-placeholder"
    />
    <div
      className="woocommerce-activity-card__subtitle is-placeholder"
    />
    <div
      className="woocommerce-activity-card__date"
    >
      <span
        className="is-placeholder"
      />
    </div>
  </div>
  <div
    className="woocommerce-activity-card__body"
  >
    <span
      className="is-placeholder"
      key="0"
    />
  </div>
  <div
    className="woocommerce-activity-card__actions"
  >
    <span
      className="is-placeholder"
    />
  </div>
</div>
`;

exports[`ActivityCardPlaceholder should render a card placeholder with date placeholder 1`] = `
<div
  aria-hidden={true}
  className="woocommerce-activity-card is-loading"
>
  <span
    className="woocommerce-activity-card__icon"
  >
    <span
      className="is-placeholder"
    />
  </span>
  <div
    className="woocommerce-activity-card__header"
  >
    <div
      className="woocommerce-activity-card__title is-placeholder"
    />
    <div
      className="woocommerce-activity-card__date"
    >
      <span
        className="is-placeholder"
      />
    </div>
  </div>
  <div
    className="woocommerce-activity-card__body"
  >
    <span
      className="is-placeholder"
      key="0"
    />
  </div>
</div>
`;

exports[`ActivityCardPlaceholder should render a card placeholder with multiple lines of content 1`] = `
<div
  aria-hidden={true}
  className="woocommerce-activity-card is-loading"
>
  <span
    className="woocommerce-activity-card__icon"
  >
    <span
      className="is-placeholder"
    />
  </span>
  <div
    className="woocommerce-activity-card__header"
  >
    <div
      className="woocommerce-activity-card__title is-placeholder"
    />
  </div>
  <div
    className="woocommerce-activity-card__body"
  >
    <span
      className="is-placeholder"
      key="0"
    />
    <span
      className="is-placeholder"
      key="1"
    />
    <span
      className="is-placeholder"
      key="2"
    />
  </div>
</div>
`;

exports[`ActivityCardPlaceholder should render a card placeholder with no content 1`] = `
<div
  aria-hidden={true}
  className="woocommerce-activity-card is-loading"
>
  <span
    className="woocommerce-activity-card__icon"
  >
    <span
      className="is-placeholder"
    />
  </span>
  <div
    className="woocommerce-activity-card__header"
  >
    <div
      className="woocommerce-activity-card__title is-placeholder"
    />
  </div>
  <div
    className="woocommerce-activity-card__body"
  />
</div>
`;

exports[`ActivityCardPlaceholder should render a card placeholder with subtitle placeholder 1`] = `
<div
  aria-hidden={true}
  className="woocommerce-activity-card is-loading"
>
  <span
    className="woocommerce-activity-card__icon"
  >
    <span
      className="is-placeholder"
    />
  </span>
  <div
    className="woocommerce-activity-card__header"
  >
    <div
      className="woocommerce-activity-card__title is-placeholder"
    />
    <div
      className="woocommerce-activity-card__subtitle is-placeholder"
    />
  </div>
  <div
    className="woocommerce-activity-card__body"
  >
    <span
      className="is-placeholder"
      key="0"
    />
  </div>
</div>
`;

exports[`ActivityCardPlaceholder should render a default placeholder 1`] = `
<div
  aria-hidden={true}
  className="woocommerce-activity-card is-loading"
>
  <span
    className="woocommerce-activity-card__icon"
  >
    <span
      className="is-placeholder"
    />
  </span>
  <div
    className="woocommerce-activity-card__header"
  >
    <div
      className="woocommerce-activity-card__title is-placeholder"
    />
  </div>
  <div
    className="woocommerce-activity-card__body"
  >
    <span
      className="is-placeholder"
      key="0"
    />
  </div>
</div>
`;
